src/main/java/org/stroganov/ws/handlers/RequestsValidator.java

```diff
@@ -0,0 +1,124 @@
+package org.stroganov.ws.handlers;
+
+import jakarta.xml.soap.*;
+import jakarta.xml.ws.handler.MessageContext;
+import jakarta.xml.ws.handler.soap.SOAPHandler;
+import jakarta.xml.ws.handler.soap.SOAPMessageContext;
+import jakarta.xml.ws.soap.SOAPFaultException;
+import org.apache.log4j.Logger;
+import org.stroganov.dao.LibraryDAO;
+import org.stroganov.entities.User;
+import org.stroganov.util.DataManager;
+
+import javax.xml.namespace.QName;
+import java.util.*;
+
+public class RequestsValidator implements SOAPHandler<SOAPMessageContext> {
+    private final Logger logger = Logger.getLogger(RequestsValidator.class);
+    private final List<String> adminRequests = Arrays.asList("ns2:deleteUser", "ns2:blockUser", "ns2:unblockUser", "ns2:getAllHistory", "ns2:addUser");
+    private User user = null;
+
+    @Override
+    public boolean handleMessage(SOAPMessageContext context) {
+        boolean isRequest = (Boolean) context.get(MessageContext.MESSAGE_OUTBOUND_PROPERTY);
+        if (!isRequest) {
+            try {
+                SOAPMessage soapMsg = context.getMessage();
+                String requestName = soapMsg.getSOAPBody().getLastChild().getNodeName();
+                if (requestName.equals("ns2:findUser")) {
+                    logger.info("Handler was missed");
+                    return true;
+                }
```

Comment:
I didn't understand the meaning of this code

Reply:
I miss first request USER. I pass it without check userLogin and password because we dont have it else.

Reply:
move "findUser" method from handler area of responsibility

Reply:
Fixed.